### ISSUE TITLE
Running multiple Vagrant processes works but takes too long.

### ISSUE THREAD

**Issue by uchagani** [mitchellh/vagrant#5122]
Environment:

* Vagrant 1.6.5
* VMWare Workstation 10
* VMWare Workstation Provider 3.2.0
* Host: Windows Server 2012 - 128GB RAM
* Guest: Windows Server 2012

In an attempt to parallelize, I can run ```vagrant up``` in multiple shells.  This works fine for the most part but takes an awful long time for the cloning.  It almost looks like vagrant is blocking the cloning part of ```vagrant up```.

![image](https://cloud.githubusercontent.com/assets/1053584/5637733/dcf8c184-95cc-11e4-9663-98b294068857.png)

It's taking almost 15 minutes to clone the VM when I run multiple vagrant processes.  If I run one at a time, it only takes a minute or 2.

**Comment by uchagani** [comment]
@mitchellh i ran some controlled tests and it looks like the piece that takes the longest is this:

```
DEBUG environment: Attempting to acquire process-lock: fpcollision

DEBUG environment: Attempting to acquire process-lock: dotlock

 INFO environment: Acquired process lock: dotlock

 INFO environment: Released process lock: dotlock

 WARN environment: Process-lock in use: fpcollision
```

Bringing up two machines at the same time in difference processes the above set up events repeats on the 2nd machine until the first has booted up.

**Comment by mitchellh** [comment]
That makes sense, those all must acquire a _process lock_ since it isn't safe to be multi-process... I don't think there is any low hanging fruit there. :(

**Comment by uchagani** [comment]
hmm.  Is this a VMWare limitation or Vagrant?  If it's vagrant, i'm wondering how some providers are able to bring up machines in parallel (or separate processes).

**Comment by mitchellh** [comment]
`fpcollision` is just a Vagrant thing where its checking for forwarding port collisions and reserving certain ports. It can't be done in parallel since Vagrant reserves some ports and if it were done in parallel two processes will likely get the same one.

**Comment by mitchellh** [comment]
(Closing, don't see a specific enough issue and therefore no real resolution right now)

**Comment by uchagani** [comment]
@mitchellh Vagrant can be sped up significantly if the cloning of the boxes is not blocked.  Allow the cloning to take place then wait for a lock on the port collision detection.

**Comment by wefner** [comment]
Did anybody got any workaround or fix maybe? Facing this very same issue after adding this line to my Vagrantfile:

`VBoxManage modifyvm "VM name" --natdnsproxy1 on`
or
`VBoxManage modifyvm "VM name" --natdnshostresolver1 on`

Deleting that didn't solve the problem since it's stuck at somewhere. I removed my `.vagrant/` directory and provision boxes again with no luck.

Now port forwarding looks messed up.

Tested with:
- Yosemite 10.10.3
- Vagrant 1.7.2
- Virtualbox 4.3.26
- Ruby 2.2.1

`vagrant up db --debug` Loads image and when it comes to boot process I can see

```
DEBUG subprocess: Waiting for process to exit. Remaining to timeout: 32000
DEBUG subprocess: Exit status: 0
 INFO warden: Calling IN action: #<Vagrant::Action::Builtin::HandleForwardedPortCollisions:0x0000010287e3c8>
DEBUG environment: Attempting to acquire process-lock: fpcollision
DEBUG environment: Attempting to acquire process-lock: dotlock
 INFO environment: Acquired process lock: dotlock
 INFO environment: Released process lock: dotlock
 WARN environment: Process-lock in use: fpcollision
```

and it keeps looping like this. It happens to 3 Vagrant projects on different folders and machine boots fine when starting it from Virtualbox itself.

Somehow `VBoxManage` parameters broke something although I don't understand why it's still failing even setting up a really basic Vagrantfile.

Can this issue be reopened?
